### DISCUSSION TITLE
Multi Ego Vehicles support in CARLA

### DISCUSSION THREAD

**Discussion by mounalbaccouch** [carla-simulator/carla#5436] [General]
Hi,

can CARLA support multi Ego Vehicles ?
I am trying to simulate a cooperative autonomous driving scenario, thus I need to simulate multiple Ego Vehicles and get their sensors data via ROS topics.

Thanks in advance.

**Comment by ralwing** [comment]
Yes it can,
You just need to use a different `role_name` e.g. hero_1, hero_2 etc. for Your vehicles. Then You'll see `/carla/hero_1/*`, /carla/hero_2/*' like topics.

**Comment by ghost** [comment]
Hello,
I am currently using caral for my research. I would like to know where I can find the corresponding configuration where I can set the different role_names?

Thanks in advance!

**Comment by ghost** [comment]
Hello @ralwing do you have any ideas?

**Comment by ralwing** [comment]
Search for it in the repository sources:
https://github.com/carla-simulator/carla/search?q=role_name

**Comment by ghost** [comment]
"While CARLA supports multi-agent environment, the Leaderboard is thought as a traffic scenario simulation around only one agent, so it doesn't support a multi-agent environment."
See also : https://github.com/carla-simulator/leaderboard/issues/132
